desc/grid.py

```diff
@@ -804,64 +1002,54 @@ def _create_nodes(  # noqa: C901
         else:
             t = np.atleast_1d(theta).astype(float)
             # enforce periodicity
-            t[t != THETA_ENDPOINT] %= THETA_ENDPOINT
+            t[t != theta_period] %= theta_period
             # need to sort to compute correct spacing
             t = np.sort(t)
             if self.sym:
-                # cut domain to relevant subdomain: delete theta > pi nodes
+                # cut domain to relevant subdomain: delete theta > π nodes
                 t = t[: np.searchsorted(t, np.pi, side="right")]
-            dt = np.zeros_like(t)
             if t.size > 1:
-                # choose dt to be the cyclic distance of the surrounding two nodes
-                dt[1:-1] = t[2:] - t[:-2]
                 if not self.sym:
-                    dt[0] = t[1] + (THETA_ENDPOINT - t[-1]) % THETA_ENDPOINT
```

Comment:
this is all just moving logic to function `_periodic_spacing()` to not have repeated code